src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -1372,10 +1380,129 @@ static void Main()
                 }
                 """;
+
+            verify(TestOptions.Regular12, expectedType12);
+            verify(TestOptions.Regular13, expectedType13);
+
+            static string getTypeParameters(string type) =>
+                type.Contains("T[]") || type.Contains("<T>") ? "<T>" : "";
+
+            static string generateMethod(string methodName, string parameterType) =>
+                $"static Type {methodName}{getTypeParameters(parameterType)}({parameterType} value) => typeof({parameterType});";
+
+            static string generateMethodSignature(string methodName, string parameterType) =>
+                $"Program.{methodName}{getTypeParameters(parameterType)}({parameterType})";
+
+            static string[] getSources(string source, string[] additionalSources)
+            {
+                var builder = ArrayBuilder<string>.GetInstance();
+                builder.Add(source);
+                builder.Add(s_collectionExtensions);
+                if (additionalSources is { }) builder.AddRange(additionalSources);
+                return builder.ToArrayAndFree();
+            }
+
+            void verify(CSharpParseOptions parseOptions, string expectedType)
+            {
+                var comp = CreateCompilation(
+                    getSources(source, additionalSources),
+                    targetFramework: TargetFramework.Net80,
+                    parseOptions: parseOptions,
+                    options: TestOptions.ReleaseExe);
+                if (expectedType is { })
+                {
+                    CompileAndVerify(comp, verify: Verification.Skipped, expectedOutput: IncludeExpectedOutput($"""
+                    {expectedType}
+                    {expectedType}
+                    """));
+                }
+                else
+                {
+                    comp.VerifyEmitDiagnostics(
+                        // 0.cs(10,17): error CS0121: The call is ambiguous between the following methods or properties: 'Program.F1(ReadOnlySpan<long>)' and 'Program.F1(ReadOnlySpan<object>)'
+                        //         var x = F1([1, 2, 3]);
+                        Diagnostic(ErrorCode.ERR_AmbigCall, "F1").WithArguments(generateMethodSignature("F1", type1), generateMethodSignature("F1", type2)).WithLocation(10, 17),
+                        // 0.cs(12,17): error CS0121: The call is ambiguous between the following methods or properties: 'Program.F2(ReadOnlySpan<object>)' and 'Program.F2(ReadOnlySpan<long>)'
+                        //         var y = F2([4, 5]);
+                        Diagnostic(ErrorCode.ERR_AmbigCall, "F2").WithArguments(generateMethodSignature("F2", type2), generateMethodSignature("F2", type1)).WithLocation(12, 17));
+                }
+            }
+        }
+
+        [Theory]
+        [InlineData("System.ReadOnlySpan<int>", "System.Span<int>", "System.ReadOnlySpan<System.Int32>")]
+        [InlineData("System.ReadOnlySpan<int>", "System.Span<object>", null)]
```

Comment:
>"System.ReadOnlySpan<int>", "System.Span<object>", null [](http://example.com/codeflow?start=20&length=55)

These cases where the collections are `ReadOnlySpan<T1>` and `Span<T2>`, or span of `T1` and array or array interface of `T2`, and where `T1` is implicitly convertible to `T2` are the breaking change cases. Should we include an `expectedType12` parameter here as well, where the test compiles against both language versions?

Reply:
Please add an item to the breaking change doc.

Reply:
> Should we include an `expectedType12` parameter here as well, where the test compiles against both language versions?

It looks like the C#12 cases are tested in `BetterConversionFromExpression_01B_CSharp12`.

Reply:
> It looks like the C#12 cases are tested in BetterConversionFromExpression_01B_CSharp12.

Yeah, I thought about trying to include it all in one big matrix, but the differences got too large and I didn't think the result was actually comprehensible. Will definitely add the whole change here to the breaking changes doc.